TestUtilities/Helpers/Decompression.swift

```diff
@@ -58,3 +58,25 @@ public struct zlib {
         }
     }
 }
+
+public struct DecompressionError: Error, CustomStringConvertible {
+    public var description: String
+}
+
+public extension URLRequest {
+    /// Returns a copy of the URLRequest with a potentially decompressed body.
+    /// - Returns: A copy of the URLRequest with a decompressed body if applicable, otherwise the original request.
+    /// - Throws: A `DecompressionError` if decompression fails.
+    func decompressed() throws -> URLRequest {
```

Comment:
This could go to Internal module. I use similar snippet to check the request body while debugging. However, its use must be avoided as it may cause memory spike in the SDK.

Reply:
If it's only for debugging, I wonder if we should make it into release code. Especially that it rather sounds like "preference" (I personally use `rum-mock-server` for debugging).